packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -997,3 +1058,137 @@ capacity provider. The weight value is taken into consideration after the base v
 }
+
+/**
+ * The options for creating an EC2 Capacity Provider.
+ */
+export interface EC2CapacityProviderProps extends AddAutoScalingGroupCapacityOptions {
+  /**
+   * The name for the capacity provider.
+   *
+   * @default CloudFormation-generated name
+   */
+  readonly capacityProviderName?: string;
+
+  /**
+   * The autoscaling group to add as a Capacity Provider.
+   */
+  readonly autoScalingGroup: autoscaling.IAutoScalingGroup;
+
+  /**
+   * Whether to enable managed scaling
+   *
+   * @default true
+   */
+  readonly enableManagedScaling?: boolean;
+
+  /**
+   * Whether to enable managed termination protection
+   *
+   * @default true
+   */
+  readonly enableManagedTerminationProtection?: boolean;
+
+  /**
+   * Maximum scaling step size. In most cases this should be left alone.
+   *
+   * @default 1000
+   */
+  readonly maximumScalingStepSize?: number;
+
+  /**
+   * Minimum scaling step size. In most cases this should be left alone.
+   *
+   * @default 1
+   */
+  readonly minimumScalingStepSize?: number;
+
+  /**
+   * Target capacity percent. In most cases this should be left alone.
+   *
+   * @default 100
+   */
+  readonly targetCapacityPercent?: number;
+}
+
+/**
+ * An EC2 Capacity Provider. This allows an ECS cluster to target a specific
+ * EC2 Auto Scaling Group for the placement of tasks. Optionally (and recommended),
+ * ECS can manage the number of instances in the ASG to fit the tasks, and can
+ * ensure that instances are not prematurely terminated while there are still tasks
+ * running on them.
+ */
+export class EC2CapacityProvider extends CoreConstruct {
+  /**
+   * Capacity provider name
+   * @default Chosen by CloudFormation
+   */
+  readonly capacityProviderName: string;
+
+  /**
+   * Auto Scaling Group
+   */
+  readonly autoScalingGroup: autoscaling.AutoScalingGroup;
+
+  /**
+   * Whether managed termination protection is enabled
+   */
+  readonly enableManagedTerminationProtection?: boolean;
+
+  constructor(scope: Construct, id: string, props: EC2CapacityProviderProps) {
+    super(scope, id);
+
+    this.autoScalingGroup = props.autoScalingGroup as autoscaling.AutoScalingGroup;
+
+    this.enableManagedTerminationProtection =
+      props.enableManagedTerminationProtection === undefined ? true : props.enableManagedTerminationProtection;
+
+    if (this.enableManagedTerminationProtection) {
+      this.autoScalingGroup.protectNewInstancesFromScaleIn();
+    }
+
+    const capacityProvider = new CfnCapacityProvider(this, id, {
+      name: props.capacityProviderName,
+      autoScalingGroupProvider: {
+        autoScalingGroupArn: this.autoScalingGroup.autoScalingGroupName,
+        managedScaling: {
+          status: props.enableManagedScaling === false ? 'DISABLED' : 'ENABLED',
+          targetCapacity: props.targetCapacityPercent || 100,
+          maximumScalingStepSize: props.maximumScalingStepSize,
+          minimumScalingStepSize: props.minimumScalingStepSize,
```

Comment:
So it looks like we're still setting these even if `props.EnabledManagedScaling` is false -- it looks like CFN is okay with that but just wanted to make sure these will be ignored/we don't need to fail early ?